### DISCUSSION TITLE
Plot reconstructed parent beta via fill and broken / bad edges

### DISCUSSION THREAD

**Discussion by accessT** [mtex-toolbox/mtex#593] [Ask Anything] [answered]
Hello together!

When i use MTEX 5.3, i get a nice plot:
![grains_ebsd_plots_5 3](https://user-images.githubusercontent.com/57536659/100292674-711dd800-2f81-11eb-88e6-228f9ac5d831.jpg)

With MTEX 5.3.1 and newer, i get bad edges:
![grains_ebsd_plots_5 3 1+](https://user-images.githubusercontent.com/57536659/100292747-ab877500-2f81-11eb-8391-7d752cf5e807.jpg)

### **Well, i found out, it has to do with the fill.m at line 17 untill 19:**
![fill_changelog](https://user-images.githubusercontent.com/57536659/100292819-de316d80-2f81-11eb-9db3-232576a8ac3f.JPG)

I just changed only this lines back like in 5.3 and it works! 
### **Is there another way to get it work like in 5.3?**

My Example:
[Probe 1 TEST Specimen 1 Site 1 Map Data 1-Subset1 - reconstruction_alpha2beta.txt](https://github.com/mtex-toolbox/mtex/files/5599710/Probe.1.TEST.Specimen.1.Site.1.Map.Data.1-Subset1.-.reconstruction_alpha2beta.txt)

```

fullpath = 'Probe 1 TEST Specimen 1 Site 1 Map Data 1-Subset1 - reconstruction_alpha2beta.txt';
mineral_ = 'Ti (beta)';
angle_ = 5;
min_grain_Size_ = 10; % Pixels
ColumnNames.Names = { 'phi1' 'Phi' 'phi2' 'x' 'y'};

CS = {'notIndexed', crystalSymmetry('m-3m', 'mineral', mineral_ ), 'notIndexed'};
ebsd = EBSD.load(fullpath,CS,'interface','generic', 'ColumnNames', ColumnNames.Names, 'Bunge');
[grains_unfiltered,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd(mineral_),'angle',angle_*degree);
grains_right=grains_unfiltered;
grains_right = grains_right(grains_right.grainSize >= min_grain_Size_); %filter
ebsd = ebsd(grains_right);
[grains_right,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd(mineral_),'angle',angle_*degree); % ReCalc

ebsd_splined = smooth(ebsd, splineFilter, 'fill'); % HERE STARTS THE PROBLEM

[grains_splined,ebsd_splined.grainId,ebsd_splined.mis2mean] = calcGrains(ebsd_splined(mineral_),'angle',angle_*degree);
grains_right2 = grains_splined(grains_splined.grainSize >= min_grain_Size_);
ebsd_splined = ebsd_splined(grains_right2);
[grains_smoothed,ebsd_splined.grainId,ebsd_splined.mis2mean] = calcGrains(ebsd_splined(mineral_),'angle',angle_*degree);
grains_smoothed = smooth(grains_smoothed, 5);
ebsd_splined = ebsd_splined(grains_smoothed);

mineral_ = string(mineral_);
fig_grains_smoothed=figure;
plot(grains_smoothed(mineral_),grains_smoothed(mineral_).meanOrientation,'micronbar','on','figSize','large','Name', 'BETA: Grains_Smoothed'); %only grain(s)
fig_ebsd_smoothed=figure;
plot(ebsd_splined(mineral_), ebsd_splined(mineral_).orientations, 'figSize', 'large','Name', 'EBSD_Smoothed');
```

**Comment by ralfHielscher** [comment, marked as answer]
Hi,

for the moment you can replace the `'none'` you marked by `'nearest'` and you should get the result you are aiming for.
In the future, I the functions `smooth` / `fill` will accept the option `'extrapolate'` which will do this.

The point is, that not everybody wants his data to be extrapolated to some rectangular boundaries.

Ralf.

**Reply by accessT** [reply]
great, thank you :)
